src/variant.rs

```diff
@@ -378,10 +475,164 @@ impl<T: ToVariant> From<T> for Variant {
     }
 }
 
+impl<T: StaticVariantType> StaticVariantType for [T] {
+    fn static_variant_type() -> Cow<'static, VariantTy> {
+        let child_type = T::static_variant_type();
+        let signature = format!("a{}", child_type.to_str());
+
+        VariantType::new(&signature).expect("incorrect signature").into()
+    }
+}
+
+impl<T: FromVariant> FromVariant for Vec<T> {
+    fn from_variant(variant: &Variant) -> Option<Self> {
+        let mut vec = vec![];
+
+        let n_children = variant.n_children();
+        if n_children != 0 {
```

Comment:
This `if` here seems redundant

Reply:
ah right, `0..0` range isn't an issue in Rust.

Reply:
In which language is it?

Reply:
I don't know actually. Forgot it, brain fart.. :)

Reply:
Ok, I was wondering what language I should try to avoid :)